CRM/Contact/BAO/Relationship.php

```diff
@@ -2427,6 +2427,9 @@ protected static function addInheritedMembership($membershipValues) {
     AND is_current_member = 1";
     $result = CRM_Core_DAO::singleValueQuery($query);
     if ($result < CRM_Utils_Array::value('max_related', $membershipValues, PHP_INT_MAX)) {
+      $values = [];
+      _civicrm_api3_custom_format_params($membershipValues, $values, 'Membership');
+      $membershipValues = array_merge($membershipValues, $values);
       CRM_Member_BAO_Membership::create($membershipValues);
```

Comment:
hmm - we should call the api here & then we would not need to format. What test cover do we have

Reply:
The api contains some code that should be moved to the BAO (according to the comments). So a direct api call executes these lines, maybe for a second time. The membership api has test coverage for updating parent and child memberships, but no tests for custom fields. So it is not surprising that they do not work.

Reply:
hmm - we so we are adding cruft because cruft exists?

Reply:
Good point, I suggest we let this PR on this place, if someone needs a quick solution for his customer. A better patch can be made if we have better test coverage for custom fields. I can create these tests but not at the moment.